setup.py

```diff
@@ -0,0 +1,149 @@
+    raise RuntimeError("Unable to find version string.")
+
+
+cmdclass = {}
+if SPHINX_AVAILABLE:
+
+    class MakeDocs(BuildDoc):
+        """Port of old `cylc-make-docs`. Removed in #2989.
+
+        This class extends the Sphinx command class for setuptools. With
+        a difference that it tries to mimic the behaviour of the previous
+        `cylc-make-docs`.
+
+        So first it will execute `make-commands.sh`, which builds the
+        commands help information, in the appendices.
+
+        Then, instead of calling one builder, this class will call the
+        builder fot he single HTML, and also the builder for the multiple
+        HTML documentation.
+
+        Finally, one more tweak in this class is to move the doctrees
+        folder (in the same level as the documentation) to within the
+        documentation folder, named `.doctrees`, as before with
+        `cylc-make-docs`.
+        """
+
+        def run(self):  # type: () -> None
+            try:
+                self.spawn(["./doc/src/custom/make-commands.sh"])
+            except DistutilsExecError as e:
+                self.warn("Failed to run make-commands.sh")
+                raise e
+            self.do_run("html", "built-sphinx")
+            self.do_run("singlehtml", "built-sphinx-single")
+
+        def do_run(self, builder: str, output_dir: str):
+            """
+            Args:
+                builder (str): name of the Sphinx builder
+                output_dir (str): directory to write the documentation produced
+            """
+            self.builder = builder
+            self.builder_target_dirs = [
+                (builder, join(self.build_dir, output_dir))]
+            super().run()
+            # move doctrees to $build_dir/.doctrees
+            correct_doctrees = join(self.builder_target_dirs[0][1],
+                                    ".doctrees")
+            rmtree(correct_doctrees, ignore_errors=True)
+            move(self.doctree_dir, correct_doctrees)
+
+    cmdclass["build_sphinx"] = MakeDocs
+
+
+install_requires = [
+    'colorama==0.4.1',
+    'isodatetime==1!2.0.0',
+    'jinja2==2.10',
+    'markupsafe==1.1.1',
+    'python-jose==3.0.1',
+    'pyzmq==18.0.1'
+]
+tests_require = [
+    'codecov==2.0.15',
+    'coverage==4.5.3',
+    'pytest-cov==2.6.1',
+    'pytest==4.4.0',
+    'pycodestyle==2.5.0',
+    'testfixtures==6.6.2'
+]
+
+extra_requires = {
+    'empy': ['EmPy==3.3.4'],
+    'docs': ['sphinx==2.0.0'],
+    'all': []
+}
+extra_requires['all'] += extra_requires['empy']
+extra_requires['all'] += extra_requires['docs']
+extra_requires['all'] += tests_require
+
+setup(
+    version=find_version("lib", "cylc", "__init__.py"),
+    long_description=open('README.md').read(),
+    long_description_content_type="text/markdown",
+    scripts=glob(join('bin', '*')),
+    packages=find_packages("lib/") + ["Jinja2Filters"],
+    package_dir={"": "lib"},
+    package_data={
+        '': ['*.txt', '*.md', '*.sh']
```

Comment:
What does `.sh*` do here?

Reply:
Files to be included in the final file uploaded to PYPI if I recall correctly. `.sh` are ignored. A previous version included css, html, etc, for cylc review (again, if memory serves me right)

Reply:
Does that mean that we are excluding files which don't end in these extensions e.g. `conf/cylc.vim`?

Reply:
Yes, that's correct. But this folder may be hidden in some venv, program files, conda, or linux lib folder.

So for the user, I think it would be easier to get this file from github or somewhere else?

The whole functional tests are also excluded. One of the reasons for pushing to run the tests with python instead of a cylc subcommand.

Do you think we should include the files under etc?

Reply:
Interesting question.  At the moment (working on 7.8.x at least) we have a command that copies `cylc import-examples` from the source directories into a location chosen by the user. 

Would it be feasible (and easy) to have something similar here, e.g. to copy everything users might need from `<cylc>/etc/` into `$HOME/.cylc/etc/`?  Just an idea...

(Because some of these files - for syntax highlighting etc. - may be needed by most users).

Reply:
>Interesting question. At the moment (working on 7.8.x at least) we have a command that copies cylc import-examples from the source directories into a location chosen by the user.

Oohh, another command that I will have to take a look.

>Would it be feasible (and easy) to have something similar here, e.g. to copy everything users might need from <cylc>/etc/ into $HOME/.cylc/<version>/etc/? Just an idea...

Hmm... will have to take a look. Because the `package_data` is for data in the package folders. Which right now are under `lib/`.

The `/tests/`, `/etc/`, etc, are not part of the package... I think we can use `data_files`. I think it was used for that, but will have to read and take a look if that works.

Reply:
(and it sounds useful.. quite sure I did that the first time I tried to use Cylc... maybe following the tutorial?)

Reply:
In the end its no big deal if users have to look elsewhere for tutorial examples, but I just thought the "install from source to user dir" idea might be worth considering.  Can punt this investigation to later work though, if you like, so we can get this PR merged.

Reply:
>Can punt this investigation to later work though, if you like, so we can get this PR merged.

+1 we will have to review most command anyway. Once `setup.py` is in, I'll have a go at #2972  too. We can make our commands `entrypoints` in setuptools. So users can even write their own `cylc brew beer` command, without touching a single line in this repository :tada: (i.e. users that need commands too specific to their site operations could still in theory extend Cylc subcommands, and I think that would be easier for us to maintain it too)

Reply:
Mmmm beeer.  That command makes me very thirsty. 🍺